lib/quicly.c

```diff
@@ -4406,13 +4475,37 @@ static int handle_handshake_done_frame(quicly_conn_t *conn, struct st_quicly_han
     return 0;
 }
 
+static int handle_ack_frequency_frame(quicly_conn_t *conn, struct st_quicly_handle_payload_state_t *state)
+{
+    quicly_ack_frequency_frame_t frame;
+    int ret;
+
+    if ((ret = quicly_decode_ack_frequency_frame(&state->src, state->end, &frame)) != 0)
+        return ret;
+
+    QUICLY_PROBE(ACK_FREQUENCY_RECEIVE, conn, probe_now(), frame.sequence, frame.packet_tolerance, frame.max_ack_delay,
+                 frame.ignore_order);
+
+    if (frame.max_ack_delay != QUICLY_LOCAL_MAX_ACK_DELAY * 1000)
```

Comment:
What's the rationale for this error?

Reply:
At the moment, we set both min_ack_delay and max_ack_delay to 25ms. Therefore, the only value that the peer can send is max_ack_delay. We are checking that.

As you have stated, I think we can open a follow-up PR for looking into using ACK_FREQUENCY to achieve loss detection signal.

Reply:
Makes sense. Can you add a comment to this effect?

Reply:
Done in ac3b36c.